src/connection/HttpConnection.ts

```diff
@@ -161,16 +161,43 @@ export default class HttpConnection extends BaseConnection {
         // if the response is compressed, we must handle it
         // as buffer for allowing decompression later
         let payload = isCompressed || isVectorTile ? new Array<Buffer>() : ''
-        const onData = isCompressed || isVectorTile
-          ? (chunk: Buffer) => { (payload as Buffer[]).push(chunk) }
-          : (chunk: string) => { payload = `${payload as string}${chunk}` }
+        const onData = isCompressed || isVectorTile ? onDataAsBuffer : onDataAsString
+
+        let currentLength = 0
+        function onDataAsBuffer (chunk: Buffer): void {
+          currentLength += Buffer.byteLength(chunk)
+          if (currentLength > maxCompressedResponseSize) {
+            // TODO: hacky solution, refactor to avoid using the deprecated aborted event
+            response.removeListener('aborted', onAbort)
+            response.destroy()
+            onEnd(new RequestAbortedError(`The content length (${currentLength}) is bigger than the maximum allowed buffer (${maxCompressedResponseSize})`))
+          } else {
+            (payload as Buffer[]).push(chunk)
+          }
+        }
+
+        function onDataAsString (chunk: string): void {
+          currentLength += Buffer.byteLength(chunk)
+          if (currentLength > maxResponseSize) {
+            // TODO: hacky solution, refactor to avoid using the deprecated aborted event
+            response.removeListener('aborted', onAbort)
```

Comment:
Yes it is, as we need to cover every other case where a request might be aborted.
The "hacky" solution is needed because of how the `aborted` event works.
In another pr we should refactor this code to decommission the `aborted` event and rely on the `close` event.
In the spirit of solving this particular issue quickly, I've preferred to do the solution described above.